packages/flutter_tools/lib/executable.dart

```diff
@@ -77,8 +77,9 @@ Future<void> main(List<String> args) async {
   final bool muteCommandLogging = (help || doctor) && !veryVerbose;
   final bool verboseHelp = help && verbose;
   final bool daemon = args.contains('daemon');
-  final bool runMachine = (args.contains('--machine') && args.contains('run')) ||
-                          (args.contains('--machine') && args.contains('attach'));
+  final bool machine = args.contains('--machine');
+  final bool runMachine = (machine && args.contains('run')) ||
+                          (machine && args.contains('attach'));
```

Comment:
this won't work. Consider for example "flutter create --machine --org run attach".

Reply:
This is true. Also, this PR doesn't change this behavior.

Reply:
Ah right this is what I filed in https://github.com/flutter/flutter/issues/75876